### PR TITLE
added validation to CreateBookRequestDto, added CustomGlobalException…

### PR DESCRIPTION
**Pull request by mmosii**
…Handler

### CODE REVIEW

File: src/main/java/mmosii/bookstore/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,13 +1,20 @@
 package mmosii.bookstore.dto;
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
```
Comment (don-bigdad) [inline src/main/java/mmosii/bookstore/dto/CreateBookRequestDto.java]: this class also can be record

**Review by don-bigdad — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -37,6 +37,10 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.hibernate.validator</groupId>
+            <artifactId>hibernate-validator</artifactId>
+        </dependency>
```
Comment (NazarSmith228) [inline pom.xml]: it is better to use `spring-boot-starter-validation`

File: src/main/java/mmosii/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -34,12 +35,13 @@ public BookDto getBookById(@PathVariable Long id) {
...
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
         return bookService.save(bookDto);
     }
 
     @PutMapping("/{id}")
-    public BookDto updateBook(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
+    public BookDto updateBook(@PathVariable Long id,
```
Comment (NazarSmith228) [inline src/main/java/mmosii/bookstore/controller/BookController.java]: just FYI

you may also validate such things as @PathVariable, @RequestParams etc. It looks like:
```
@PathVariable @Positive Long id
```

In order to be able to perform such validation you need to annotate your controller class with @Validated annotation

again, this is just FYI - apply it or not :)

File: src/main/java/mmosii/bookstore/dto/BookDto.java [hunk 4]
```diff
@@ -1,15 +1,7 @@
...
-public class BookDto {
-    private Long id;
-    private String title;
-    private String author;
-    private String isbn;
-    private BigDecimal price;
-    private String description;
-    private String coverImage;
+public record BookDto(Long id, String title, String author, String isbn,
+                      BigDecimal price, String description, String coverImage) {
```
Comment (NazarSmith228) [inline src/main/java/mmosii/bookstore/dto/BookDto.java]: usually fields of records are formatted as follows:
```
public record BookDto(Long id, 
                      String title, 
                      String author, 
                      String isbn,
                      BigDecimal price, 
                      String description, 
                      String coverImage) {
```

File: src/main/java/mmosii/bookstore/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (NazarSmith228) [inline src/main/java/mmosii/bookstore/dto/CreateBookRequestDto.java]: why not record?

File: src/main/java/mmosii/bookstore/dto/CreateBookRequestDto.java [hunk 5]
```diff
@@ -1,13 +1,20 @@
...
+    @NotNull
```
Comment (NazarSmith228) [inline src/main/java/mmosii/bookstore/dto/CreateBookRequestDto.java]: it is better to use `@NotBlank` with strings

File: src/main/java/mmosii/bookstore/dto/CreateBookRequestDto.java [hunk 6]
```diff
@@ -1,13 +1,20 @@
...
     private String title;
+    @NotNull
     private String author;
+    @NotNull
     private String isbn;
+    @NotNull
+    @Min(0)
```
Comment (NazarSmith228) [inline src/main/java/mmosii/bookstore/dto/CreateBookRequestDto.java]: `@Min(0)` looks weird :)

could you use `@Positive` or `@PositiveOrZero`?

File: src/main/java/mmosii/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
```
Comment (NazarSmith228) [inline src/main/java/mmosii/bookstore/exception/CustomGlobalExceptionHandler.java]: time is pretty useless in response body - it does not bring any value to the client

status is obvious as well - no sense to have it in body

File: src/main/java/mmosii/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
```
Comment (NazarSmith228) [inline src/main/java/mmosii/bookstore/exception/CustomGlobalExceptionHandler.java:21]: do you have any other exceptions except for `MethodArgumentNotValidException`?

does it make sense to handle them as well?

**Review by NazarSmith228 — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution

**Review by liliia-ponomarenko — APPROVED** [review]
Great!

**Pull request merged**